src/main/java/mate/academy/Main.java

```diff
@@ -2,25 +2,28 @@
 
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 public class Main {
+    public static final int THREADS_COUNT = 5;
+
     public static void main(String[] args) {
-        // Feel free to play with AsyncRequestProcessor in this main method if you want
-        ExecutorService executor = null; // Provide implementation that fits your needs
-        AsyncRequestProcessor asyncRequestProcessor = new AsyncRequestProcessor(executor);
+        ExecutorService executor = Executors.newFixedThreadPool(THREADS_COUNT);
+        try {
+            AsyncRequestProcessor asyncRequestProcessor = new AsyncRequestProcessor(executor);
 
-        // Simulating multiple concurrent requests
-        String[] userIds = {"user1", "user2", "user3", "user1"}; // Note: "user1" is repeated
-        CompletableFuture<?>[] futures = new CompletableFuture[userIds.length];
+            String[] userIds = {"user1", "user2", "user3", "user1"};
+            CompletableFuture<?>[] futures = new CompletableFuture[userIds.length];
```

Comment:
I think we can move this out of the try block, the smaller they are - the better